tests/posix/mutex/src/posix_mutex.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <errno.h>
+#include <pthread.h>
+
+#define STACK_SIZE 1024
+
+static K_THREAD_STACK_DEFINE(stack, STACK_SIZE);
+
+#define SLEEP_MS 100
+
+pthread_mutex_t mutex1;
+pthread_mutex_t mutex2;
+
+void *foo(void *p1)
+{
+	int i, rc;
+
+	/* Sleep for maximum 300 ms as main thread is sleeping for 100 ms */
+
+	for (i = 0; i < 3; i++) {
+		rc = pthread_mutex_trylock(&mutex1);
+		if (rc == 0) {
+			break;
+		}
+		k_sleep(SLEEP_MS);
+	}
+
+	zassert_false(rc, "try lock failed\n");
+	TC_PRINT("mutex lock is taken\n");
+	zassert_false(pthread_mutex_unlock(&mutex1),
+		      "mutex unlock is falied\n");
```

Comment:
in this and all asserts below and above, remove the "\n" from the message, this is now being added by default.